crates/hotfix/src/session.rs

```diff
@@ -120,7 +120,7 @@ struct Session<M, S> {
     store: S,
     awaiting_test_response: Option<TestRequestId>,
 
-    heartbeat_timer: Pin<Box<Sleep>>,
+    heartbeat_timer: Option<Pin<Box<Sleep>>>,
```

Comment:
This is the actual change, and it doesn't really do that much, but I wanted to get it out as a separate PR as it has the new approach for testing flows involving passage of time using tokio's `advance`.